inspektor/inspector.py

```diff
@@ -77,9 +77,12 @@ def is_python(self):
         return self.is_script(language='python')
 
     def is_toignore(self):
-        for pat in self.ignore:
-            if fnmatch.fnmatch(self.path, pat):
+        path = self.path
+        if path.startswith('./'):
+            path = path[2:]
+        for pattern in self.ignore:
+            if fnmatch.fnmatch(path, pattern):
                 return True
-            if self.path.startswith(os.path.abspath(pat)):
+            if pattern in self.path:
```

Comment:
Well this is a bit too lenient to me and could lead to unwanted skips. For example:

```
$ tree
.
├── data
└── src
    ├── data
    └── datadir.py
```

Without this patch: `--exclude data` checks `src/*` as well as `src/data/*`, but with this patch it does not actually check any files as `data` is part of `datadir.py` name. Can you please elaborate a bit on your usecase?